pkg/providers/cloudstack/cloudstack.go

```diff
@@ -530,7 +531,7 @@ func AnyImmutableFieldChanged(oldCsdc, newCsdc *v1alpha1.CloudStackDatacenterCon
 	if oldCsmc.Spec.ComputeOffering != newCsmc.Spec.ComputeOffering {
 		return true
 	}
-	return false
+	return !reflect.DeepEqual(oldCsmc.Spec.UserCustomDetails, newCsmc.Spec.UserCustomDetails)
```

Comment:
Comments in my previous PR mentioned that deepequal is not recommended in non-test code.
May write a customer equal method for usercustomerdetails or loop through its keys to compare. Currently, we may not need to sort the keys before camparing.

Reply:
I can replace with
```	
        if len(oldCsmc.Spec.UserCustomDetails) != len(newCsmc.Spec.UserCustomDetails) {
		return true
	}
	for key, value := range oldCsmc.Spec.UserCustomDetails {
		if value != newCsmc.Spec.UserCustomDetails[key] {
			return true
		}
	}
	return false
```